gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java

```diff
@@ -48,6 +49,8 @@
   // Used for single-byte reads.
   private final byte[] singleReadBuf = new byte[1];
 
+  private GoogleCloudStorageItemInfo itemInfo;
```

Comment:
Why do we need to save this as a field?

Reply:
Will this be of use for us when we have to reuse itemInfo for the same session? @medb 
In the same session, in case someone needs to use it, we can check if the existing Path and the path of the itemInfo are the same and reuse it, instead of getting the itemInfo which will fire a new request.

Reply:
I think that this is extra optimization that we do not need necessary do now, let's focus on core functionality first.

Reply:
@medb Ok, sure